### DISCUSSION TITLE
Avro vs. Parquet?

### DISCUSSION THREAD

**Discussion by vab2048** [holixon/axon-avro-serializer#81] [Q&A]
Asking out of pure curiosity (not because I think one is better than the other).

Why Avro over other formats such as parquet?
Anything special Avro has that parquet is missing?

**Comment by zambrovski** [comment]
Hi,

I think there is not better or worse, it is a question of the ecosystem. Decent big data frameworks are specialized to store parquet, and AVRO is an old dinosaur in the "binary encoding market".

What we were looking at was to support a concise format (binary) to store the data and have a native schema support. We believe that usage of a **schema-first approach** is a good idea if you design format of your events for (an infinite) storage of events in event store. Especially, the schema reference and schema evolution are important features for event lifecycle.

In doing so we profit from Avro single-object encoding (to store a reference of the schema instead of a whole schema) and introduce an abstraction of a Schema registry, using adapters for common schema registries (like in-memory, or Apicurio or you might implement your own). In addition, the Avro feature of schema compatiblity (and incompatiblity with detection of certain types) are very helpful for design of event upcasters. We implemented a converter to a specific typed record for conversion between bytes and a designated type used in the application and make use of conversion between bytes and generic type record (which is in general a map<string, object>) to use the latter as intermediate representation, needed for event upcasting....

These were our ideas,

Cheers,

Simon
